Casks/xmind-zen.rb

```diff
@@ -1,15 +1,18 @@
 cask "xmind-zen" do
-  version "11.1.1-202110191930"
-  sha256 "d3e9b88061f3f6af650ed91388eeb950a13e41d7e8876848d91f0e6f6228b23a"
+  version "11.1.2,202111071943"
+  sha256 "923ca09d1b81435088680179c50f73bb27342b2a5e97e16d0daaa0433aea5de3"
 
-  url "https://dl3.xmind.net/XMind-for-macOS-#{version}.dmg"
+  url "https://dl3.xmind.net/XMind-for-macOS-#{version.before_comma}-#{version.after_comma}.dmg"
   name "XMind"
   desc "Mindmap and brainstorming app"
   homepage "https://www.xmind.net/desktop/"
 
   livecheck do
     url "https://www.xmind.net/zen/download/mac/"
-    strategy :header_match
+    regex(/[._-]v?(\d+(?:\.\d+)+)[._-](\d+)\.dmg/i)
+    strategy :header_match do |headers, regex|
+      headers["location"].match(regex)&.then { |match| "#{match[1]},#{match[2]}" }
+    end
```

Comment:
```suggestion
    regex(/XMind-for-macOS[._-]v?(\d+(?:\.\d+)+)[._-](\d+)\.dmg/i)
    strategy :header_match do |headers, regex|
      match = headers["location"].match(regex)
      next if match.blank?

      "#{match[1]},#{match[2]}"
    end
```

The suggestion above brings the regex more in line with common patterns for matching a version from a filename. Namely, the regex generally covers the entire filename, including the leading text before the `[._-]` (i.e., `XMind-for-macOS[._-]`).

Past that, the commentary around `&.then` in the [`spotify` review](https://github.com/Homebrew/homebrew-cask/pull/114328#discussion_r749474940) applies here as well. Despite the terseness of using `#then`, I would prefer to continue using the existing format for now, for the previously-outlined reasons.

Reply:
Is there a reason for using `XMind-for-macOS` instead of `XMind[._-]for[._-]macOS`?

Reply:
The short explanation is that `XMind[._-]for[._-]macOS` is technically more flexible but it's generally overkill in practice. With livecheck regexes, we try to find a balance between strictness/looseness (explicit vs. implicit), verbosity/brevity, etc., so inevitably there are some tradeoffs.

Over the course of creating/modifying/reviewing thousands of `livecheck` blocks (primarily in formulae), I didn't see any notable benefit to additional flexibility in the text that precedes `[._-]v?(\d+(?:\.\d+)+)` in the vast majority of cases. Substituting `[._-]` in the leading text only becomes necessary when upstream is shown to be inconsistent in this part of the filename, so we [ideally] only do this on a case-by-case basis. Off the top of my head, there are probably only a handful of existing instances (if any) that are technically necessarily.

This standard was originally developed in formula `livecheck` blocks but part of the idea behind this approach in casks is that the leading filename text is hardcoded in the `url`, so it kind of makes sense to hardcode it in the `livecheck` block regex as well (to keep them aligned). A filename change would affect both, so it would potentially be a problem from the perspective of automated version bump PRs if livecheck successfully returned a version and it was substituted into the `url` only to fail. [In the future, I plan to make changes that allow us to return the full filename (where possible) in the JSON output, so we'll be able to sidestep this issue in some instances.]

There's a case to be made for using a more flexible regex in the `livecheck` block that wouldn't break in this context but I'm currently of the opinion that it may be better for `livecheck` to fail if the leading text changes. At least in homebrew/core, I keep an eye on the failures (especially those that use a `livecheck` block) and it acts as a visible signal that changes need to be made.

That said, I'm not married to this idea and I may change course over time as I continue improving livecheck. This is simply what I standardized on in the past and I haven't seen much value in switching to a more verbose format without a compelling reason. I almost never see a regex fail to match a filename in such a way that using `[._-]` in the leading text would have avoided the problem. I _have_ seen the delimiter change between the leading text and version (e.g., `something_1.2.3` to `something-1.2.3`), as well as in some suffixes (e.g., `1.2.3_4` to `1.2.3-4`), and that's why we use `[._-]` in those areas.